Introduce Time type for bar chart

JeanMarcMilletScality opened the pull request:
Enhance Barchart component with time-based data formatting and custom ticks.

Add centered text option to ConstrainedText component. 
Update storybook with time range examples for Barchart.

JeanMarcMilletScality committed:
Enhance Barchart component with time-based data formatting and custom ticks.
Add centered text option to ConstrainedText component.
Update storybook with time range examples for Barchart.

bert-e commented:
# Hello jeanmarcmilletscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/core-ui/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @hervedombya

* @jeanmarcmilletscality

* @jbwatenbergscality

* @chengyanjin

JBWatenbergScality commented on src/lib/components/barchartv2/Barchart.component.tsx:
```diff
@@ -108,20 +111,214 @@ const formatPrometheusDataToChartData = (
+
+    return {
+      dayKey,
+      category: `${dayName}${dayNumber}${monthName}`,
```
nit: Might be missing spaces

JBWatenbergScality commented on src/lib/components/barchartv2/Barchart.component.tsx:
```diff
@@ -108,20 +111,214 @@ const formatPrometheusDataToChartData = (
+
+      const existingData = dayMap.get(dayKey)!;
+      // Set value for the day (no summing needed since there's only one per day)
+      existingData[dataKey] = value;
```
Not sure about the use of this line of code. It doesn't change the dayMap or does it by ref ?

JeanMarcMilletScality replied:
Map.get is a ref to the object so it modify it

JeanMarcMilletScality committed:
Refactor format function:
Format function can handle category or time types
Introduce date format function for bar  chart in FormattedDateTime

JeanMarcMilletScality commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue ARTESCA-15250.

Goodbye jeanmarcmilletscality.




*The following options are set:* **approve**
